mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoImplementationGenerator.java

```diff
@@ -103,6 +105,22 @@ public String addPreparedStatement(
     return fieldName;
   }
 
+  @Override
+  public String addQueryProvider(
+      ExecutableElement methodElement,
+      TypeMirror providerClass,
+      List<ClassName> entityHelperTypes) {
+    // Invokers are not shared between methods, so always generate a new name
+    String fieldName = nameIndex.uniqueField(methodElement.getSimpleName().toString() + "Invoker");
```

Comment:
Nice, i was wondering if this would work for overloaded methods and it appears it does (i.e. `findSliceInvoker1` is defined if a previous `findSliceInvoker` exists.

Reply:
Yes. Same for prepared statements.